test-packages/test-services-e2e/v4/test-service/TestEntity50PropApi.ts

```diff
@@ -603,7 +605,8 @@ export class TestEntity50PropApi<
         DECIMAL_PROPERTY_2: fieldBuilder.buildEdmTypeField(
           'DecimalProperty2',
           'Edm.Decimal',
-          true
+          true,
+          9
```

Comment:
[q] What is the max allowed precision as per odata? Just want to make sure we're testing the correct values. If I remember cds allows only till 7

Reply:
From OData spec:
>For a decimal value: the maximum number of significant decimal digits of the property’s value; it MUST be
a positive integer.
>For a temporal value (datetime-with-timezone-offset, duration, or time-of-day): the number of decimal
places allowed in the seconds portion of the value; it MUST be a non-negative integer between zero and
twelve. 

Currently, `precision` values get generated for both temporal and decimal values.
But, I only use it for formatting `DateTimeOffset` fields.
I am okay with generating them for decimal fields as well, eventhough we don't use it, do you feel otherwise?

Reply:
I think it would be nicer if we generate it for types where we actually support it. otherwise it might be confusing.

Reply:
I would only update our test client code for cases where we actually support the behavior.

Reply:
It was updated automatically now during generation, as we always add `precision` if it's available in the EDMX file.
Let me check, how to restrict this during generation only for temporal fields.

Reply:
I have restricted to generate `precision` into  `buildEdmTypeField ` only if the EDM type is DateTimeOffset.